src/main/java/com/thebois/models/beings/AbstractBeing.java

```diff
@@ -1,42 +1,75 @@
 package com.thebois.models.beings;
 
-import java.util.Random;
+import java.util.Objects;
 
-import com.thebois.models.IDeepClonable;
 import com.thebois.models.Position;
+import com.thebois.models.beings.roles.AbstractRole;
+import com.thebois.models.beings.roles.RoleFactory;
 
 /**
  * An abstract implementation of IBeing.
  */
-public abstract class AbstractBeing implements IBeing, IDeepClonable<IBeing> {
+public abstract class AbstractBeing implements IBeing {
 
-    // The current position held by the AbstractBeing.
-    private Position currentPosition;
+    // The max speed of the AbstractBeing
+    private static final float MAX_WALKING_DISTANCE = 0.1f;
     // The position the AbstractBeing wants to reach.
     private Position destination;
-    // The max speed of the AbstractBeing
-    private final float maxWalkingDistance = 5;
-    // temporary: n*n world size
-    private final int worldSize = 500;
-    private Random random = new Random();
-
-    /**
-     * Creates an instance of AbstractBeing.
-     */
-    public AbstractBeing() {
-        this.destination = new Position();
-        this.currentPosition = new Position();
-    }
+    // The current position held by the AbstractBeing.
+    private Position currentPosition;
+    private AbstractRole role;
 
     /**
      * Creates an AbstractBeing with an initial position.
      *
      * @param currentPosition the initial position of the AbstractBeing.
      * @param destination     the initial destination of the AbstractBeing.
      */
-    public AbstractBeing(Position currentPosition, Position destination) {
+    public AbstractBeing(final Position currentPosition, final Position destination) {
         this.currentPosition = currentPosition;
         this.destination = destination;
+        this.role = RoleFactory.idle();
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getDestination(), getPosition(), getRole());
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) return true;
+        if (!(o instanceof AbstractBeing)) return false;
+        final AbstractBeing that = (AbstractBeing) o;
+        return Objects.equals(getDestination(), that.getDestination())
+               && Objects.equals(currentPosition,
+                                 that.currentPosition)
+               && Objects.equals(getRole(), that.getRole());
+    }
+
+    protected Position getDestination() {
+        return destination.deepClone();
+    }
+
+    @Override
+    public Position getPosition() {
+        return currentPosition.deepClone();
+    }
+
+    @Override
+    public AbstractRole getRole() {
+        if (role == null) return null;
```

Comment:
Nice catch, I did a search for all `null`s in the project, but must've missed this one!

Reply:
Since we never should use null as a role, I'm removing the null check entirely so we can see the errors whenever someone accidentally does set a role to null.